utils.cpp

```diff
@@ -275,41 +275,57 @@ std::vector<gpt_vocab::id> gpt_tokenize(const gpt_vocab & vocab, const std::stri
     return tokens;
 }
 
+// TODO: Calculate this constant from the vocabulary
+#define MAX_TOKEN_LEN 10
```

Comment:
Max is actually 16!

```
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
----------------
▁Einzelnachweise
****************
================
################
addEventListener
////////////////
▁implementations
▁straightforward
▁characteristics
▁representations
................
________________
▁Representatives
▁transformations
```

There are also a bunch more:


<details>
<summary>15</summary>

```
▁implementation
springframework
includegraphics
▁representation
▁authentication
▁classification
▁transformation
OnClickListener
printStackTrace
▁characteristic
▁administration
▁initialization
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
▁estadounidense
▁administrative
▁Unterscheidung
▁interpretation
▁internationale
▁representative
▁multiplication
▁simultaneously
▁principalmente
▁configurations
▁responsibility
▁Classification
▁Administration
▁Fußballspieler
▁Schriftsteller
▁veröffentlicht
▁setContentView
▁Authentication
▁Mediabestanden
PropertyChanged
▁recommendation
backgroundColor
▁identification
```

</details>


<details>
<summary>14</summary>

```
▁▁▁▁▁▁▁▁▁▁▁▁▁▁
▁International
▁documentation
▁configuration
▁international
▁automatically
▁corresponding
ViewController
getElementById
▁understanding
▁functionality
▁Unfortunately
▁communication
AndroidRuntime
▁circumstances
▁approximately
▁Championships
▁unfortunately
ActivityThread
eltemperaturen
▁Massachusetts
Authentication
▁approximation
▁significantly
▁Internacional
▁Bibliographie
amerikanischer
▁Entertainment
disambiguation
▁distributions
▁consideration
▁universitaire
▁internacional
▁Alternatively
▁autorytatywna
▁distinguished
▁contributions
▁Configuration
adratkilometer
implementation
▁relationships
▁specification
preventDefault
▁investigation
▁partiellement
Longrightarrow
authentication
▁административ
▁contradiction
longrightarrow
▁développement
▁possibilities
▁compatibility
▁Transfermarkt
▁collaboration
▁establishment
▁verschiedenen
▁independently
▁organizations
▁StringBuilder
▁notifications
▁decomposition
▁modifications
LayoutInflater
RelativeLayout
CollectionView
▁concentration
▁computational
▁administrator
▁participation
▁extraordinary
▁französischen
▁Genomsnittlig
▁authorization
▁environmental
▁stackoverflow
▁neighbourhood
▁interpolation
```

</details>

I am a little bit worried we may run into “glitch tokens” like GPT-3 (but that’s something that can be explored in the future!)

Reply:
Thanks! I didn't realize there were tokens like that . I changed it to 18 with intention to calculate this from the vocabulary in the future.